src/components/TweetItem.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faHeart as faEmptyHeart } from "@fortawesome/free-regular-svg-icons";
+import { faHeart as faSolidHeart } from "@fortawesome/free-solid-svg-icons";
+import { connect } from 'react-redux';
+import { likeTweet } from '../actions/tweetsActions'
+
+
+function TweetItem(props) {
+    const [isLiked, setIsLiked] = useState(props.isLiked);
+
+    const handleLike = (id) => {
+        setIsLiked(!isLiked);
+        if (!isLiked) props.onLike({ id: id, isLiked: !isLiked });
```

Comment:
Still not sure what, but something here feels unclear and can be done in simpler way.
Let's discuss as part of second CR batch